midend/parserUnroll.cpp

```diff
@@ -367,6 +366,41 @@ class ParserSymbolicInterpreter {
             auto mc = sord->to<IR::MethodCallStatement>();
             auto e = ev.evaluate(mc->methodCall, false);
             success = reportIfError(state, e);
+        } else if (auto bs = sord->to<IR::BlockStatement>()) {
+            IR::IndexedVector<IR::StatOrDecl> newComponents;
+            for (auto* component : bs->components) {
+                auto newComponent = executeStatement(state, component, valueMap);
+                if (!newComponent)
+                    success = false;
+                else
+                    newComponents.push_back(newComponent);    
+            }
+            sord = new IR::BlockStatement(newComponents);
+        } else if (sord->is<IR::IfStatement>()) {
+            auto ifs = sord->to<IR::IfStatement>()->clone();
+            auto ifcond = ev.evaluate(ifs->condition, true);
+            success = reportIfError(state, ifcond);
+            if (success) {
+                const IR::Statement* ifComponent = nullptr;
+                if (ifcond != nullptr) {
+                    if (ifcond)
+                        ifComponent = ifs->ifTrue;
+                    else 
+                        ifComponent = ifs->ifFalse;
+                    auto newComponent = executeStatement(state, ifComponent, valueMap);
+                    if (!newComponent) {
+                        success = false;
```

Comment:
What does success = false mean here?

Reply:
We check if newComponent is nullptr, similar to the actions in checks for AssignmentStatement, since it may end up so that some part of the block will not be processed and nullptr will be returned